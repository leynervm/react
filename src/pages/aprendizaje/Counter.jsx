import { useState } from 'react';
import PropTypes from 'prop-types';

const Counter = ({ start, end }) => {

    const [counter, setCounter] = useState(start)
    const handleIncrement = () => setCounter(counter + 1)
    const handleDecrement = () => setCounter(counter - 1)
    const handleReset = () => setCounter(start)

    return (
        <>
            <div>
                <div>
                    MAX
                    <span>{end}</span>
                </div>
                <h1 data-testid="test-counter">{counter}</h1>

                <div className="flex justify-center gap-1 mt-6 w-full">
                    <button aria-label='btn-disminuir' className='link-navigation-aprendizaje' type='button' onClick={handleDecrement}>
                        Disminuir
                    </button>
                    <button aria-label='btn-incrementar' className='link-navigation-aprendizaje' type='button' onClick={handleIncrement}>
                        Incrementar
                    </button>
                    <button aria-label='btn-reset' className='bg-red-500 !border-red-500 !ring-red-500 !text-white link-navigation-aprendizaje' type='button' onClick={handleReset}>
                        Resetear
                    </button>
                </div>
            </div>
        </>
    )
}

Counter.propTypes = {
    start: PropTypes.number.isRequired,
    end: PropTypes.number.isRequired,
};

export default Counter