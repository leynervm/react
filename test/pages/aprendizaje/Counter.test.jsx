import { afterEach, describe, expect, test } from "vitest";
import Counter from "../../../src/pages/aprendizaje/Counter";
import { cleanup, fireEvent, render, screen } from "@testing-library/react";


describe('TEST COUNTER EN COMPONENT en <Counter />', () => {
    afterEach(cleanup)

    const start = 100
    test('debe de hacer match con el snapshoot', () => {
        const { container } = render(<Counter />);
        expect(container).toMatchSnapshot();
    })

    test('debe mostrar el valor inicial de 100', () => {
        render(<Counter start={start} />);
        expect(screen.getByText(start).innerHTML).toContain(start);
    })

    test('debe incrementar con el boton Incrementar +1', () => {
        render(<Counter start={start} />);
        fireEvent.click(screen.getByRole('button', { name: 'btn-incrementar' }));
        // screen.debug();
        expect(screen.getByText('101')).toBeTruthy();
    })

    test('debe disminuir con el boton Disminuir -1', () => {
        render(<Counter start={start} />)
        fireEvent.click(screen.getByRole('button', { name: 'btn-disminuir' }))
        // screen.debug()
        expect(screen.getByText('99')).toBeTruthy();
    })

    test('debe funcionar el boton Resetear', () => {
        render(<Counter start={start} />)
        fireEvent.click(screen.getByRole('button', { name: 'btn-incrementar' }))
        fireEvent.click(screen.getByRole('button', { name: 'btn-incrementar' }))
        fireEvent.click(screen.getByRole('button', { name: 'btn-incrementar' }))
        fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }))
        expect(screen.getByText(start)).toBeTruthy();
    })

})