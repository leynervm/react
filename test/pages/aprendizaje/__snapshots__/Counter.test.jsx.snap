// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TEST COUNTER EN COMPONENT en <Counter /> > debe de hacer match con el snapshoot 1`] = `
<div>
  <div>
    <div>
      MAX
      <span />
    </div>
    <h1
      data-testid="test-counter"
    />
    <div
      class="flex justify-center gap-1 mt-6 w-full"
    >
      <button
        aria-label="btn-disminuir"
        class="link-navigation-aprendizaje"
        type="button"
      >
        Disminuir
      </button>
      <button
        aria-label="btn-incrementar"
        class="link-navigation-aprendizaje"
        type="button"
      >
        Incrementar
      </button>
      <button
        aria-label="btn-reset"
        class="bg-red-500 !border-red-500 !ring-red-500 !text-white link-navigation-aprendizaje"
        type="button"
      >
        Resetear
      </button>
    </div>
  </div>
</div>
`;
